### PR TITLE
[v3.0] Add newer Python 3 versions to build matrix

### PR DESCRIPTION
**Pull request by adamantike**
For the `3.0` branch, add newer Python versions to the build matrix, up
to Python 3.7.

This is motivated by #1100, which backported a fix for the `3.0` branch
to work with Python 3.7.

### CODE REVIEW

**Comment by adamantike** [comment]
@auvipy ping for awareness! With these changes to the `3.0` branch, only the `pypy` build is currently failing in Travis, and everything from AppVeyor. Let me know if it makes sense to keep working on them, before putting more effort into it.

Hopefully, this could allow us to have another 3.x release, compatible with newer Python versions :)

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
-      uname -a
-      lsb_release -a
-      sudo pip install tox
-script: tox -v -e $TRAVIS_PYTHON_VERSION -- -v
+sudo: false
+cache: pip
+
+matrix:
+  include:
+    - python: 2.6
```
Comment (auvipy) [inline .travis.yml]: this should  be removed

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+      env: TOXENV=py26 TOX_VERSION="<3" TOX_DEPS="pluggy<0.6.0" # Latest supported version
+      dist: trusty
+    - python: 2.7
+      env: TOXENV=py27
+    - python: 3.3
+      env: TOXENV=py33 TOX_VERSION="<3" TOX_DEPS="pluggy<0.6.0 virtualenv<15.2.0" # Latest supported version
+      dist: trusty
+    - python: 3.4
```
Comment (auvipy) [inline .travis.yml]: remove this

File: .travis.yml [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+cache: pip
+
+matrix:
+  include:
+    - python: 2.6
+      env: TOXENV=py26 TOX_VERSION="<3" TOX_DEPS="pluggy<0.6.0" # Latest supported version
+      dist: trusty
+    - python: 2.7
+      env: TOXENV=py27
+    - python: 3.3
```
Comment (auvipy) [inline .travis.yml]: drop 3.3

**Comment by adamantike** [comment]
I would gladly remove those versions! As it would be a breaking change for those Python versions, I wasn't sure of the approach you take on this project, as this commit will be tagged as minor/patch within the `3.0` branch. If that's OK, I will proceed with those changes.

**Comment by auvipy** [comment]
It's normal to drop the python version reaching EOL. we are not supposed to support celery 3.x as well but doing so for the sake of easier migration to 4.x versions.

**Comment by adamantike** [comment]
AppVeyor is now fixed by using almost the same configuration as `master`. The only pending one is the `pypy` environment in Travis, which of course, passes when running tests locally.

**Comment by auvipy** [comment]
can you update pypy2.7 to 7.2?

**Comment by adamantike** [comment]
I don't see `pypy2.7-7.2` still supported by Travis, considering its [nightly builder](https://github.com/travis-ci/travis-nightly-builder/blob/734c0330f7d8ac4aa18df211c93a39fd5e93414c/Rakefile#L83), and its [documentation](https://docs.travis-ci.com/user/languages/python/#python-versions). That's why I'm using the same PyPy versions we test in `master`.

**Comment by auvipy** [comment]
I added it on celery master, check it

**Comment by adamantike** [comment]
I can add it to check if it works, but that environment didn't run for your PR: https://travis-ci.org/celery/celery/builds/601834152, and I don't see it running afterwards because it's in the `lint` stage, which is only executed if the previous stage is successful, while it's been failing consistently in `master` celery.

**Comment by adamantike** [comment]
I tested both `pypy2.7-7.2` and `pypy2.7-7.2.0`, and Travis fails with:
```
Downloading archive: https://storage.googleapis.com/travis-ci-language-archives/python/binaries/ubuntu/16.04/x86_64/pypy2.7-7.2.tar.bz2

0.13s$ curl -sSf -o pypy2.7-7.2.tar.bz2 ${archive_url}

curl: (22) The requested URL returned error: 404 Not Found

Unable to download pypy2.7-7.2 archive. The archive may not exist. Please consider a different version.
```
(from https://travis-ci.org/celery/kombu/jobs/602304400)

**Comment by auvipy** [comment]
OK stick with 7.1.x

**Comment by adamantike** [comment]
So, I still don't know why the `pypy` environment fails in Travis, where this simple assertion about installed codecs fails:
```
import codecs
codecs.lookup('ascii')
```

My guess is that the PyPy installation provided by Travis is missing something, but I don't think it should block this PR from being merged.

This pull request is now ready for review. After this one, I will check if migrating the test suite to PyTest makes any difference to fix that breaking environment.

**Comment by adamantike** [comment]
@auvipy, let me know if is there anything else I should add to this PR, to be merged.

**Pull request merged**